google/resource-snippets/pubsub-v1/test_alternatives.sh

```diff
@@ -0,0 +1,35 @@
+set -e
+
+TF_PROJECT_ID={TF_PROJECT_ID}
+DM_PROJECT_ID={DM_PROJECT_ID}
+
+
+gcloud auth login
+
+# Create DM resources
+gcloud config set project $DM_PROJECT_ID
+gcloud services enable deploymentmanager.googleapis.com
+gcloud deployment-manager deployments create d1 --config pubsub.yaml
+
+# Create Terraform resources
+gcloud config set project $TF_PROJECT_ID
```

Comment:
This comment still applies. You added

```
gcloud auth application-default login
export GOOGLE_APPLICATION_CREDENTIALS=~/.config/gcloud/application_default_credentials.json
```
after setting the project for DM, but if the project changes for TF (`gcloud config set project $TF_PROJECT_ID`) then we need to do it again since application default credentials are per project.

Maybe I should create a `test_alternatives.sh` example where we only try to use one project to simplify these auth scenarios, for future PRs.